### PR TITLE
Nabla page refresh required after currency swap

### PR DESCRIPTION
**Pull request by Sharqiewicz**
🦈 🟢 PR
---
After swapping one currency to another on the platform, users must refresh the page to initiate another swap. If they attempt a second swap without refreshing, they encounter the following error: "Transaction failed."

**Issue**: Token Approval is not reset after successful swap.
**Solution**: Implement approval query invalidation after successful swap.

### CODE REVIEW

**Comment by Sharqiewicz** [comment]
@ebma @prayagd @TorstenStueber Ready for review

**Comment by Sharqiewicz** [comment]
@ebma Do you approve ✅?

**Review by ebma — APPROVED** [review]
Code looks good to me, nice job @Sharqiewicz 👍

File: src/helpers/contracts.ts [hunk 1]
```diff
@@ -14,15 +14,9 @@ export interface ContractBalance {
...
-export function parseContractBalanceResponse(decimals: number, balanceResponse: INumber): ContractBalance;
 export function parseContractBalanceResponse(
-  decimals: number | undefined,
-  balanceResponse: INumber | undefined,
-): ContractBalance | undefined;
-
-export function parseContractBalanceResponse(
-  decimals: number | undefined,
-  balanceResponse: INumber | undefined,
+  decimals?: number,
```
[the remark this answers is not in the document]
Comment (Sharqiewicz) [reply src/helpers/contracts.ts:18]: @TorstenStueber I think implicit checks make the behaviour more predictable and minimise the possibility of runtime errors, but after a while I agree that overloading the type leverages the Typescript strengths and makes the code more consistent. I'll revert this change

**Review by TorstenStueber — APPROVED** [review]
👍

**Pull request merged**